.deployment/deploy.yml

```diff
@@ -0,0 +1,210 @@
+---
+# vim: et:ts=2:sw=2:sts=2
+
+- hosts: all
+
+  vars:
+    # default branch
+    # will usually be opverwritten by --extra-vars='branch=<breanch>'
+    branch: master
+    b64_branch: "{{ branch  | b64encode | replace('=', '') }}"
+    id: "{{ branch | regex_replace('[^a-zA-Z0-9]', '') }}-{{ b64_branch }}"
```

Comment:
> This doesn't necessarily make the id unique, right? Multiple PRs can be create from the same branch (at a later time).

It's unique for the Ansible `branch` variable which does not have to be the git branch. I want this to be the git branch for pushes later. That also will overwrite previous deployments (basically advancing the head). For pull requests I would set this either to the pull request ID or possibly a date and ID if we really want a deployment per update which I'm not sure we want since that's actually pretty expensive (deployments need far more resources than with e.g. Studio)

> And maybe hash the thing instead of base64 encoding to get a constant length?

Hashing to a constant length actually makes this non-unique ;-P
But sure, you can use a reasonable strong hash to minimize the risk.
For example, we could use:

    {{ 'xy' | password_hash('sha256', 'mysecretsalt') }}

That would give you URLs like:

- https://xy-769a4e6d0003189c7e96c5d9b7e810a0d11c3a12832527ec94b0f86d277f51ca.tobira.opencast.org

I think, I prefer

- https://xy-eHk.tobira.opencast.org

Reply:
Yeah ok, good points.

- I also prefer a shorter id, so base64 is fine
- PR number as id for PRs seems like a good idea
- I'm fine with overwriting previous deployments for the same branch/PR. Only keep the latest for a specific branch/PR.